x-pack/plugins/fleet/server/services/epm/packages/get.ts

```diff
@@ -142,6 +150,90 @@ export async function getPackages(
   return packageListWithoutStatus;
 }
 
+export async function getInstalledPackages(options: {
+  savedObjectsClient: SavedObjectsClientContract;
+  type?: PackageDataStreamTypes;
+  nameQuery?: string;
+  pageAfter?: SortResults;
+  pageSize: number;
+  sortDirection: 'asc' | 'desc';
+}) {
+  const { savedObjectsClient, pageAfter, pageSize, nameQuery, sortDirection, type } = options;
+
+  const packageSavedObjects = await savedObjectsClient.find<Installation>({
```

Comment:
this function could be broken down to smaller functions for readability and unit tested.